### PR TITLE
kubectl ko performance enhance

### PR DESCRIPTION
**Pull request by changluyi**
1. add result format
2. add gc
3. add svc performance test
4. add diagnose to node port

- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Features
- Bug fixes
- Docs
- Tests

### Which issue(s) this PR fixes:
Fixes #[2959](https://github.com/kubeovn/kube-ovn/issues/2959)

### <samp>🤖 Generated by Copilot at 2908995</samp>

This pull request enhances the pinger component to support custom target protocol, IP and port for connectivity check. It adds a new flag `--target-ip-ports` to the pinger binary and updates the `pinger` package to use the `TargetIPPorts` field of the `Configuration` struct. It also fixes some minor typos in the log messages.

### <samp>🤖 Generated by Copilot at 2908995</samp>

> _`pinger` can probe_
> _target protocol and port_
> _autumn network test_

### <samp>🤖 Generated by Copilot at 2908995</samp>

*  Add a new field `TargetIPPorts` to the `Configuration` struct to store the target protocol, IP and port for connectivity check ([link](https://github.com/kubeovn/kube-ovn/pull/2975/files?diff=unified&w=0#diff-56f139df697aad97197df93aedd368692be48a0ef3d6a4399b7f17fea0c6c8fcR56))
*  Add a new flag `--target-ip-ports` to the `ParseFlags` function to parse the user input for the target protocol, IP and port, and assign it to the `argTargetIPPorts` variable ([link](https://github.com/kubeovn/kube-ovn/pull/2975/files?diff=unified&w=0#diff-56f139df697aad97197df93aedd368692be48a0ef3d6a4399b7f17fea0c6c8fcR76))
*  Assign the value of `argTargetIPPorts` to the `TargetIPPorts` field of the `Configuration` struct returned by the `ParseFlags` function ([link](https://github.com/kubeovn/kube-ovn/pull/2975/files?diff=unified&w=0#diff-56f139df697aad97197df93aedd368692be48a0ef3d6a4399b7f17fea0c6c8fcR135))
*  Add a new function `checkAccessTargetIPPorts` to the `pinger` package to perform the connectivity check for the target protocol, IP and port specified by the `TargetIPPorts` field of the `Configuration` struct ([link](https://github.com/kubeovn/kube-ovn/pull/2975/files?diff=unified&w=0#diff-ef5e19d2a06e783042497957187c9a7df962e6db67629dfbe8075e6f12c02704R272-R314))
*  Call the `checkAccessTargetIPPorts` function from the `ping` function if the `TargetIPPorts` field is not empty, and set the `errHappens` flag to true if any error occurs ([link](https://github.com/kubeovn/kube-ovn/pull/2975/files?diff=unified&w=0#diff-ef5e19d2a06e783042497957187c9a7df962e6db67629dfbe8075e6f12c02704R71-R76))
*  Fix typos in the log messages of the `pingNodes` and `pingPods` functions, changing "connnectivity" to "connectivity" ([link](https://github.com/kubeovn/kube-ovn/pull/2975/files?diff=unified&w=0#diff-ef5e19d2a06e783042497957187c9a7df962e6db67629dfbe8075e6f12c02704L97-R112), [link](https://github.com/kubeovn/kube-ovn/pull/2975/files?diff=unified&w=0#diff-ef5e19d2a06e783042497957187c9a7df962e6db67629dfbe8075e6f12c02704L168-R184))

### CODE REVIEW

File: dist/images/kubectl-ko [hunk 1]
```diff
@@ -599,6 +629,26 @@ diagnose(){
...
 
+  quitDiagnose() {
+    kubectl delete ds -l app=$CONN_CHECK_LABEL -n $KUBE_OVN_NS
+    kubectl delete svc -l app=$CONN_CHECK_LABEL -n $KUBE_OVN_NS
+  }
+  trap quitDiagnose EXIT
+
+  applyTestNodePortService
+
+  targetIPPorts=""
```
Comment (oilbeater) [inline dist/images/kubectl-ko:640]: 这些需要手填的变量要放到脚本前面，方便统一填写，并在注释里给出填写的例子

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: dist/images/kubectl-ko (hunk 1 above)
Comment (changluyi) [reply dist/images/kubectl-ko:640]: 这个参数targetIPPorts只是初始化 不是给用户填写的，

我看额外加个命令 kubectl ko diagnose targetIPPorts {xxx} 给用户。

Comment (changluyi) [reply dist/images/kubectl-ko]: fixed

**Comment by oilbeater** [comment]
需要在 e2e 里增加新命令的测试

**Review by oilbeater — APPROVED** [review]

**Pull request merged**